File: Jack.Presentation/Controllers/API/TesteController.vb
Imports System.Net
Imports System.Web.Http

Namespace Controllers.API
    <RoutePrefix("api/Teste")>
    Public Class TesteController
        Inherits ApiController

        ' GET: api/Teste
        '<Route("~/")>
        '<HttpGet()>
        Public Function GetValues() As IEnumerable(Of String)
            Return New String() {"value1", "value2"}
        End Function

        ' GET: api/Teste/5
        '<Route("~/{ID:int}")>
        '<HttpGet()>
        Public Function GetValue(ByVal id As Integer) As String
            Return "value"
        End Function

        ' POST: api/Teste
        Public Sub PostValue(<FromBody()> ByVal value As String)

        End Sub

        ' PUT: api/Teste/5
        Public Sub PutValue(ByVal id As Integer, <FromBody()> ByVal value As String)

        End Sub

        ' DELETE: api/Teste/5
        Public Sub DeleteValue(ByVal id As Integer)

        End Sub
    End Class
End Namespace

File: Jack.Presentation.Web/Controllers/API/TesteController.cs
using Jack.Model;
using System.Collections.Generic;
using System.Web.Http;

namespace Controllers.API
{
    public class TesteController : ApiController
	{

		// GET: api/Teste
		public IList<CriancaMoralCrista> GetValues()
		{
            IList<CriancaMoralCrista> criancas;
            Jack.Business.CriancaMoralCrista oCrianca = new Jack.Business.CriancaMoralCrista();

            criancas = oCrianca.LoadAll();

            return criancas;
        }

		// GET: api/Teste/5
		public string GetValue(int id)
		{
			return "value";
		}

		// POST: api/Teste
		public void PostValue(	[FromBody()] string value)
		{
		}

		// PUT: api/Teste/5
		public void PutValue(int id, [FromBody()] string value)
		{
		}

		// DELETE: api/Teste/5

		public void DeleteValue(int id)
		{
		}
	}
}
